remill/Arch/X86/Semantics/CALL_RET.cpp

```diff
@@ -91,7 +91,40 @@ DEF_SEM(IRETD) {
   state.hyper_call = AsyncHyperCall::kX86IRet;
   return memory;
 }
-}
 
 DEF_ISEL(IRETD_32) = IRETD;
+
+#elif ADDRESS_SIZE_BITS == 64
+DEF_SEM(IRETQ) {
+  auto new_rip = PopFromStack<uint64_t>(memory, state);
+  auto new_cs = static_cast<uint16_t>(PopFromStack<uint64_t>(memory, state));
+  auto temp_rflags = PopFromStack<uint64_t>(memory, state);
+  Flags f = {};
+  f.flat = temp_rflags;
+  Write(REG_PC, new_rip);
+  Write(REG_CS.flat, new_cs);
+  state.rflag = f;
+  state.aflag.af = f.af;
+  state.aflag.cf = f.cf;
+  state.aflag.df = f.df;
+  state.aflag.of = f.of;
+  state.aflag.pf = f.pf;
+  state.aflag.sf = f.sf;
+  state.aflag.zf = f.zf;
+  state.hyper_call = AsyncHyperCall::kX86IRet;
+
+  // update hidden part (segment descriptor) of cs
+
+  auto new_rsp = PopFromStack<uint64_t>(memory, state);
+  auto new_ss = static_cast<uint16_t>(PopFromStack<uint64_t>(memory, state));
+  Write(REG_RSP, new_rsp);
+  Write(REG_SS.flat, new_ss);
+
+  // update hidden part (segment descriptor) of ss
+
```

Comment:
Can you file an Issue for this? Also, put this as someting like:

```c++
// TODO(tathanhdinh): Update hidden part (segment descriptor) of ss,
//                    see Issue #NNN.
```

My guess is that you'll want to do the segment update by a `__remill_sync_hyper_call`?

Reply:
@pgoodman thanks you for the comment.

> My guess is that you'll want to do the segment update by a `__remill_sync_hyper_call`?
I think we can make the update be explicit (see #334).

Reply:
> Can you file an Issue for this? Also, put this as someting like:

Sure, PR updated.